src/lib.rs

```diff
@@ -1,26 +1,132 @@
 //! A framework for the modular construction and evaluation of metaheuristics.
 //!
-//! TODO: Write more about core goals, architecture, and give examples.
+//! MAHF allows easy construction and experimental analysis of metaheuristics by
+//! decomposing them into their fundamental components.
+//!
+//! The framework supports not only evolutionary algorithms, but also any other
+//! metaheuristic frameworks, including non-population-based,
+//! constructive, and especially hybrid approaches.
+//!
+//! # Overview
+//!
+//! MAHF aims to make the construction and modification of metaheuristics as simple and reliable as possible.
+//! In addition to construction it also provides utilities for logging, evaluation, and comparison of those heuristics:
+//!
+//! - Simple modular construction of metaheuristics
+//! - State management and state tracking
+//! - Collection of common operators
+//! - Templates for common heuristics
+//! - Flexible logging of runtime information
```

Comment:
This reads a bit weird.

Reply:
All of it or specific bullet points?

Reply:
First the sentence which lists some features, then the colon without any transition and lastly the list which also lists features.

Reply:
Is the new version better?